Symfony/CS/Fixer/PSR1/EncodingFixer.php

```diff
@@ -30,13 +31,13 @@ public function __construct()
     /**
      * {@inheritdoc}
      */
-    public function fix(\SplFileInfo $file, $content)
+    public function fix(\SplFileInfo $file, Tokens $tokens)
     {
+        $content = $tokens->generateCode();
+
         if (0 === strncmp($content, $this->BOM, 3)) {
-            return substr($content, 3);
+            $tokens->setCode(substr($content, 3));
```

Comment:
couldn't this be done on the Tokens collection directly rather than forcing reparsing the content ?